CHANGELOG: update for v0.27.0

pks-t opened the pull request:
I feel like this should be the most important changes since v0.26.0.

carlosmn commented on CHANGELOG.md:
```diff
@@ -6,26 +6,89 @@ v0.26 + 1
   to any repository (and does not apply configuration like 'insteadof' rules).
   This is mostly useful for e.g. emulating `git ls-remote` behavior.
 
+* `git_diff_patchid` lets you generate patch IDs for diffs.
```
I think this should be `git_diff_patchid()` for consistency.

carlosmn commented on CHANGELOG.md:
```diff
@@ -6,26 +6,89 @@ v0.26 + 1
+* Fix for reading patches with whitespace in the compared files' paths.
+
+* Fix for writing only the last refspec into FETCH_HEAD in case multiple
+  refspecs are being fetched.
```
This makes it sound like the fix we did is that only the last refspec is written, when it's the other way around.  Maybe something like

> We now fill `FETCH_HEAD` from all passed refspecs instead of overwriting with the last one.

pks-t commented:
Thanks @carlosmn, used both your improvements

pks-t commented:
Rebased and added another batch of changes which have been merged since the last time. Also, I moved the new conflict marker length option entry to "API additions", as we are not in fact breaking existing APIs but only extending the options structure.
